### PR TITLE
[Feature][Connectors-v2] Adding Connectors for SingleStoreFeature singlestore

### PR DESCRIPTION
**Pull request by 852511489**
# Completed Modifications
## Issue 1 & 5 & 10 (MAJOR/CRITICAL) - SingleStoreDialect and Compatibility
**SingleStoreDialect.java**:
- Added class-level JavaDoc to clarify:
  - Reuses MySQL behaviors: type mapping/RowConverter, Upsert, SHOW TABLE STATUS/CRC32, `rewriteBatchedStatements`, and Schema changes (DDL)
  - Explicitly noted that setting `converterName` of RowConverter to "MySQL" is a deliberate compatibility design
  - Added `@see` references for `MysqlDialect`, `MysqlJdbcRowConverter`, and `MySqlTypeConverter`
  - Supplemented JavaDoc for the `fieldIde` constructor parameter

## Issue 2 (MINOR) - RowConverter Displays MYSQL
**SingleStoreDialectTest.java**:
- Added JavaDoc to `testRowConverter` explaining that SingleStore intentionally reuses MySQL's RowConverter; "MySQL" may appear in logs, but the current dialect remains SingleStore.

## Issue 3 (MINOR) - Dependency Version Documentation
**pom.xml**:
- Added comments to `singlestore.jdbc.version` explaining the rationale for selecting version 1.2.8 (MySQL protocol compatibility, JDBC 4.2 compliance) and recommending regular checks for security and compatibility updates.

**Documentation**:
- In the English Source documentation (under "Dependency Usage"), noted that version 1.2.8 is used for build and testing; other versions require self-validation.

## Issue 4 (MINOR) - URL Format Validation
**SingleStoreDialectFactory.java**:
- Enhanced URL format validation in `acceptsURL`:
  - Requires URL format: `jdbc:singlestore:[loadbalance:|sequential:]//host...` (non-empty host must exist after `//`)
  - Rejects invalid formats (e.g., `jdbc:singlestore://`, `jdbc:singlestore:///db`, `jdbc:singlestore:loadbalance://`)

**SingleStoreDialectTest**:
- Added `testSingleStoreDialectFactoryRejectsMalformedUrl` and assertions for several valid URL patterns.

## Issue 6 (MAJOR) - Integration Testing
**Documentation (en/zh)**:
- Added "Manual integration testing" section to the Source documentation:
  - Clarified that the Testcontainers SingleStore image is not used currently
  - Provided steps to validate with real SingleStore 7.1+ instances:
    1. Create databases/tables
    2. Run source/sink jobs
    3. Verify sharding and upsert functionality
    4. Optional: Schema Evolution testing

## Issue 7 (MINOR) - Boundary Condition Unit Tests
**SingleStoreDialectTest.java**:
- Added new test cases:
  - `testQuoteIdentifierWithNullFieldIde`: `null` `fieldIde` falls back to the original identifier
  - `testQuoteIdentifierWithEmptyFieldIde`: Empty `fieldIde` falls back to the original identifier
  - `testQuoteIdentifierWithValidFieldIde`: Valid `fieldIde` (LOWERCASE/UPPERCASE)
  - `testQuoteIdentifierWithInvalidFieldIde`: Throws `IllegalArgumentException` for invalid `fieldIde`

## Issue 8 (MINOR) - Version Documentation
**docs/en & docs/zh**:
- Updated "SingleStore v7.1+" to explicitly state: "Tested on version 7.1 and later"
- Briefly explained this version range is required for JDBC and MySQL-compatible SQL (e.g., `SHOW TABLE STATUS`, `CRC32`, `ON DUPLICATE KEY UPDATE`); earlier versions are not officially supported.

## Issue 9 (MINOR) - Troubleshooting
**docs/en/connectors/source/SingleStore.md**:
- Added "FAQ / Troubleshooting" table covering:
  - Connection issues
  - Driver problems
  - Sharding/sampling issues
  - Upsert/batch processing issues
  - Schema change issues

**docs/zh/connectors/source/SingleStore.md**:
- Added corresponding "常见问题与排查 (FAQ / Troubleshooting)" section.

**Sink Documentation (en/zh)**:
- Added references to the Source documentation's FAQ section, plus brief sink-specific troubleshooting notes.

## Issue 11 (MAJOR) - Schema Changes
**SingleStoreDialect.java**:
- In the JavaDoc reuse list, explicitly added: "Schema change (DDL) – ALTER TABLE ADD/MODIFY/DROP COLUMN inherits from MySQL; validate on SingleStore if using Schema Evolution".

**Documentation FAQ**:
- In the "Schema evolution (ALTER TABLE) issues" section, explained that DDL inherits from MySQL; recommended testing on the target SingleStore version and documenting any discrepancies.

### CODE REVIEW

**Comment by DanielCarter-stack** [comment]
### Issue 1: fieldIde access permission inconsistency

**Location**:
`seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/singlestore/SingleStoreDialect.java:28`

**Code**:
```java
public SingleStoreDialect(String fieldIde) {
    super(fieldIde);  // Call parent class constructor
}
```

**Related context**:
```java
// MysqlDialect.java:55-60
public String fieldIde = FieldIdeEnum.ORIGINAL.getValue();  // ⚠️ public field

public MysqlDialect() {}

public MysqlDialect(String fieldIde) {
    this.fieldIde = fieldIde;  // Direct assignment
}

// StarRocksDialect.java:23-27
public StarRocksDialect(String fieldIde) {
    this.fieldIde = fieldIde;  // Direct access to parent class field, not super(fieldIde)
}
```

**Problem description**:
SingleStoreDialect uses `super(fieldIde)` to call the parent class constructor, while StarRocksDialect directly accesses `this.fieldIde = fieldIde`. Both approaches are correct, but the style is inconsistent. This indicates that there are different inheritance pattern practices in the project.

**Potential risks**:
- Low risk: Both approaches are functionally equivalent
- Maintenance risk: Developers may be confused about which pattern should be used

**Impact scope**:
- Direct impact: SingleStoreDialect's constructor implementation
- Indirect impact: Code style inconsistency with StarRocksDialect
- Affected area: Single Connector (SingleStore)

**Severity**: MINOR

**Improvement suggestion**:
Keep the current implementation (`super(fieldIde)`), as this is the approach that better aligns with object-oriented design principles. Consider standardizing the project style during code review, but this is not a blocking issue for this PR.

---

### Issue 2: converterName displaying "MySQL" may cause log confusion

**Location**:
`seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/singlestore/SingleStoreDialectTest.java:53-56`

**Code**:
```java
@Test
public void testRowConverter() {
    SingleStoreDialect dialect = new SingleStoreDialect();
    Assertions.assertNotNull(dialect.getRowConverter());
    Assertions.assertEquals(
            DatabaseIdentifier.MYSQL, dialect.getRowConverter().converterName());
            // ⚠️ Assert converterName is "MySQL"
}
```

**Related context**:
```java
// SingleStoreDialect inherits from MysqlDialect
// MysqlDialect.getRowConverter() returns new MysqlJdbcRowConverter()

// MysqlJdbcRowConverter's converterName is hardcoded
```

**Problem description**:
Since SingleStoreDialect reuses MysqlJdbcRowConverter, the converter's converterName is fixed as "MySQL". Although "MySQL" rather than "SingleStore" will be displayed in logs and monitoring, this is expected behavior (as documented in JavaDoc and test comments). However, this may cause the following confusion:
1. When users see "MySQL converter" in logs, they may mistakenly think they are connecting to MySQL rather than SingleStore
2. Converter performance for SingleStore and MySQL cannot be distinguished in monitoring metrics

**Potential risks**:
- User confusion: Names in logs and monitoring are inconsistent with the actual database being used
- Difficult debugging: When troubleshooting issues, one may mistakenly think it's a MySQL connector problem

**Impact scope**:
- Direct impact: All users using the SingleStore connector
- Indirect impact: Log analysis, monitoring systems
- Affected area: SingleStore Connector

**Severity**: MINOR (already documented)

**Improvement suggestion**:
```java
// Solution 1: Override getRowConverter() in SingleStoreDialect
@Override
public JdbcRowConverter getRowConverter() {
    // Return a wrapper that modifies converterName but keeps logic unchanged
    return new MysqlJdbcRowConverter() {
        @Override
        public String converterName() {
            return DatabaseIdentifier.SINGLESTORE;
        }
    };
}

// Solution 2: Modify MysqlJdbcRowConverter constructor to accept converterName parameter
// But this affects existing code, not recommended
```

**Rationale**:
Although the current implementation is functionally correct and documented, from an observability perspective, converterName should reflect the actual dialect being used. It is recommended to consider this issue in future refactoring, but this is not a blocking issue for this PR.

---

### Issue 3: URL validation regex does not validate port number validity

**Location**:
`seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/singlestore/SingleStoreDialectFactory.java:35-38`

**Code**:
```java
private static final Pattern URL_PATTERN =
    Pattern.compile(
        "^jdbc:singlestore:(?:loadbalance:|sequential:)?//([^/?#]+)(/[^?#]*)?.*");
```

**Related context**:
```java
// Accepted URL examples:
// jdbc:singlestore://host:3306/db
// jdbc:singlestore://host:abc/db  ⚠️ Port number is invalid, but regex doesn't validate
```

**Problem description**:
The URL_PATTERN regular expression uses `([^/?#]+)` to match the host part, which includes the port number. However, it does not validate whether the port number is a valid number. For example:
- `jdbc:singlestore://host:abc/db` will be accepted (regex match succeeds)
- `jdbc:singlestore://host:99999/db` will be accepted (port exceeds range)

These issues rely on the JDBC driver to throw exceptions when connecting.

**Potential risks**:
- Delayed error discovery: Configuration errors are exposed at runtime rather than at startup
- Unclear error messages: JDBC driver errors may not clearly indicate URL format issues

**Impact scope**:
- Direct impact: URL format validation logic
- Indirect impact: User experience (error timing)
- Affected area: SingleStore Connector

**Severity**: MINOR

**Improvement suggestion**:
```java
private static final Pattern URL_PATTERN =
    Pattern.compile(
        "^jdbc:singlestore:(?:loadbalance:|sequential:)?//" +
        "([^/?#]+)" +                          // host:port part
        "(/[^?#]*)?.*");

@Override
public boolean acceptsURL(String url) {
    if (url == null || !url.startsWith(URL_PREFIX)) {
        return false;
    }
    Matcher m = URL_PATTERN.matcher(url);
    if (!m.matches()) {
        return false;
    }
    String hostPart = m.group(1);
    if (hostPart == null || hostPart.trim().isEmpty()) {
        return false;
    }
    
    // Added: Validate port number
    String[] parts = hostPart.split(",");
    for (String part : parts) {
        String hostPort = part.trim();
        if (hostPort.contains(":")) {
            String[] hp = hostPort.split(":");
            if (hp.length != 2) return false;
            try {
                int port = Integer.parseInt(hp[1]);
                if (port < 1 || port > 65535) return false;
            } catch (NumberFormatException e) {
                return false;
            }
        }
    }
    
    return true;
}
```

**Rationale**:
Although the JDBC driver will ultimately validate connection parameters, performing basic validation at the dialect layer can provide earlier and clearer error feedback. However, this increases code complexity and requires handling multi-host (loadbalance) scenarios. The current implementation is already sufficient because:
1. The JDBC driver will catch invalid ports
2. Common errors (like empty host) are already caught
3. Port number errors are rare

It is recommended to keep the current implementation and only document in the docs that the JDBC driver will validate connection parameters.

---

### Issue 4: Missing URL format coverage tests for JdbcDialectLoaderTest

**Location**:
`seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialectLoaderTest.java:43-52`

**Code**:
```java
@Test
public void shouldFindSingleStoreDialect() throws Exception {
    JdbcDialect jdbcDialect =
            JdbcDialectLoader.load("jdbc:singlestore://localhost:3306/test", null, "");
    Assertions.assertInstanceOf(SingleStoreDialect.class, jdbcDialect);
}

@Test
public void shouldFindSingleStoreDialectByDialect() throws Exception {
    JdbcDialect jdbcDialect = JdbcDialectLoader.load("jdbc:other://host/db", "SingleStore", "");
    Assertions.assertInstanceOf(SingleStoreDialect.class, jdbcDialect);
}
```

**Related context**:
```java
// SingleStoreDialectTest has detailed URL validation tests
testSingleStoreDialectFactoryAcceptsUrl()
testSingleStoreDialectFactoryRejectsMalformedUrl()

// But JdbcDialectLoaderTest only tests normal URLs, not abnormal URLs
```

**Problem description**:
`shouldFindSingleStoreDialect` in `JdbcDialectLoaderTest` only tests valid URLs. Although URL validation logic has been tested in `SingleStoreDialectTest`, from an integration testing perspective, `JdbcDialectLoaderTest` should cover the following scenarios:
1. Whether invalid URL formats are correctly rejected
2. Whether dialect parameter override works properly
3. Whether URL prefix matching is correct

**Potential risks**:
- Incomplete integration test coverage
- If JdbcDialectLoader's logic changes, it may not be detected in time

**Impact scope**:
- Direct impact: JdbcDialectLoaderTest test suite
- Indirect impact: Code regression risk
- Affected area: Overall test quality

**Severity**: MINOR

**Improvement suggestion**:
```java
@Test
public void shouldRejectInvalidSingleStoreUrl() throws Exception {
    // Test that invalid URLs should not load SingleStoreDialect
    JdbcDialect jdbcDialect = JdbcDialectLoader.load("jdbc:singlestore://", null, "");
    // Should return GenericDialect or throw an exception
    Assertions.assertNotInstanceOf(SingleStoreDialect.class, jdbcDialect);
}

@Test
public void shouldLoadSingleStoreDialectByDialectName() throws Exception {
    // Test dialect parameter override
    JdbcDialect jdbcDialect = JdbcDialectLoader.load(
        "jdbc:unknown://host/db", "SingleStore", "LOWERCASE");
    Assertions.assertInstanceOf(SingleStoreDialect.class, jdbcDialect);
    // Verify fieldIde parameter is passed correctly
    Assertions.assertEquals("`col`", jdbcDialect.quoteIdentifier("COL"));
}
```

**Rationale**:
Although `SingleStoreDialectTest` already covers URL validation logic, `JdbcDialectLoaderTest` should conduct integration testing from the loader's perspective. It is recommended to supplement these test cases, but this is not a blocking issue for this PR.

---

### Issue 5: Missing scope declaration for singlestore-jdbc-client in pom.xml

**Location**:
`seatunnel-connectors-v2/connector-jdbc/pom.xml:400-403`

**Code**:
```xml
<dependency>
    <groupId>com.singlestore</groupId>
    <artifactId>singlestore-jdbc-client</artifactId>
</dependency>
```

**Related context**:
```xml
<!-- 在 dependencyManagement 中声明了 scope=provided -->
<dependency>
    <groupId>com.singlestore</groupId>
    <artifactId>singlestore-jdbc-client</artifactId>
    <version>${singlestore.jdbc.version}</version>
    <scope>provided</scope>
</dependency>

<!-- 但在 dependencies 中未显式声明 scope -->
<!-- 依赖会继承 dependencyManagement 的配置 -->
```

**Problem description**:
In `dependencyManagement`, the scope of `singlestore-jdbc-client` is set to `provided`. When referenced in the `dependencies` section, the scope is not explicitly declared. Although Maven will inherit the dependencyManagement configuration, best practice is to also explicitly declare scope in dependencies to improve readability and avoid accidental overrides.

**Potential risks**:
- Low risk: Maven's dependency transitivity mechanism ensures the scope is correct
- Maintenance risk: If someone modifies dependencyManagement in the future, they may forget to synchronize changes to dependencies

**Impact scope**:
- Direct impact: pom.xml dependency declaration
- Indirect impact: None (Maven will handle correctly)
- Affected area: Build configuration

**Severity**: MINOR (code style issue)

**Improvement suggestion**:
```xml
<dependency>
    <groupId>com.singlestore</groupId>
    <artifactId>singlestore-jdbc-client</artifactId>
    <scope>provided</scope>  <!-- 显式声明 scope -->
</dependency>
```

**Rationale**:
Explicitly declaring scope can improve code readability, allowing other developers to see at a glance that this is a provided dependency. However, this is not a functional issue, and Maven will handle dependencies correctly.

---

**Comment by 852511489** [comment]
Addresses code review feedback for the SingleStore JDBC connector with minor improvements to observability, validation, tests, and build config.

## Changes

- **converterName**: Override `getRowConverter()` in `SingleStoreDialect` to return a converter whose `converterName()` is `DatabaseIdentifier.SINGLESTORE` instead of "MySQL", so logs and metrics reflect SingleStore usage.
- **URL validation**: In `SingleStoreDialectFactory.acceptsURL()`, validate that the port (when present) is a numeric value in 1–65535; reject invalid ports (e.g. `host:abc`, `host:0`, `host:65536`) and support multi-host URLs (comma-separated).
- **Tests**:  
  - Update `SingleStoreDialectTest.testRowConverter()` to assert `converterName()` is SINGLESTORE; add URL tests for invalid port formats.  
  - In `JdbcDialectLoaderTest`, add `shouldRejectInvalidSingleStoreUrl()` (invalid URL should not load SingleStore dialect) and `shouldLoadSingleStoreDialectByDialectNameWithFieldIde()` (dialect name override with fieldIde).
- **Build**: Explicitly set `<scope>provided</scope>` for `singlestore-jdbc-client` in `connector-jdbc/pom.xml` for clarity.

No change to the existing `SingleStoreDialect(String fieldIde)` constructor (keeps `super(fieldIde)`).

**Comment by 852511489** [comment]
@DanielCarter-stack I‘m sorry, but is there anything else I need to change?

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/singlestore/SingleStoreDialect.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    public JdbcRowConverter getRowConverter() {
+        return new MysqlJdbcRowConverter() {
+            @Override
+            public String converterName() {
+                return DatabaseIdentifier.SINGLESTORE;
+            }
+        };
+    }
```
Comment (yzeng1618) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/singlestore/SingleStoreDialect.java:66]: Anonymous classes create a new instance on every call and are not serializable, which can cause issues in distributed execution.Consider extracting the anonymous class into a named

File: docs/en/connectors/source/SingleStore.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+## Example
+
+### Read by table_path
+
+```hocon
+source {
+  Jdbc {
+    url = "jdbc:singlestore://localhost:3306/test"
+    driver = "com.singlestore.jdbc.Driver"
+    user = "root"
```
Comment (yzeng1618) [inline docs/en/connectors/source/SingleStore.md]: Parameter inconsistency

File: docs/en/connectors/source/SingleStore.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+### Read by query
+
+```hocon
+source {
+  Jdbc {
+    url = "jdbc:singlestore://localhost:3306/test"
+    driver = "com.singlestore.jdbc.Driver"
+    user = "root"
```
Comment (yzeng1618) [inline docs/en/connectors/source/SingleStore.md]: ditto

File: docs/en/connectors/source/SingleStore.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+```
+
+### With connection parameters
+
+```hocon
+source {
+  Jdbc {
+    url = "jdbc:singlestore://localhost:3306/test?rewriteBatchedStatements=true"
+    driver = "com.singlestore.jdbc.Driver"
+    user = "root"
```
Comment (yzeng1618) [inline docs/en/connectors/source/SingleStore.md]: ditto

File: seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/singlestore/SingleStoreDialectTest.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+        Assertions.assertFalse(factory.acceptsURL("jdbc:singlestore:loadbalance://"));
+        Assertions.assertFalse(factory.acceptsURL("jdbc:singlestore://host:abc/db"));
+        Assertions.assertFalse(factory.acceptsURL("jdbc:singlestore://host:0/db"));
+        Assertions.assertFalse(factory.acceptsURL("jdbc:singlestore://host:65536/db"));
+    }
+
+    @Test
+    public void testSingleStoreDialectFactoryDialectName() {
+        SingleStoreDialectFactory factory = new SingleStoreDialectFactory();
+        Assertions.assertEquals(DatabaseIdentifier.SINGLESTORE, factory.dialectFactoryName());
```
Comment (yzeng1618) [inline seatunnel-connectors-v2/connector-jdbc/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/singlestore/SingleStoreDialectTest.java:117]: missing validation for the SINGLESTORE constant

**Comment by yzeng1618** [comment]
Thanks for the PR. There are still some CI-related issues that need to be addressed:
- plugin-mapping.properties needs to be updated
- The pom.xml in seatunnel-dist needs to be updated
- The CI label needs to be added in label-scope-conf
Please update the PR after fixing these issues. Thanks.

**Review by DanielLeens — CHANGES_REQUESTED** [review]
Thank you for the detailed follow-up and for sticking with this as a first contribution. I pulled the current branch locally and the overall direction looks good.

I still see one blocking issue before I would merge it:

`SingleStoreDialectFactory.acceptsURL()` now rejects valid bracketed IPv6 URLs when no explicit port is provided. For example, a URL like `jdbc:singlestore://[2001:db8::1]/db` reaches `lastIndexOf(':')`, treats the last colon inside the IPv6 literal as a port separator, and then rejects `1]` as an invalid port. Because dialect discovery depends on `acceptsURL()`, those jobs would fail before the JDBC driver even gets a chance to handle the URL.

I think the safe fix is either:
- handle bracketed IPv6 hosts explicitly when parsing the host part, or
- relax this validation back to only rejecting clearly empty / malformed host segments and leave detailed host parsing to the driver.

The rest of the cleanup in this PR is moving in a good direction, so once this URL-validation regression is fixed I will be happy to take another look.

**Comment by davidzollo** [comment]
I pushed a follow-up commit to the PR branch to address the remaining `acceptsURL()` blocker on the current code path.

What changed:
- fixed `SingleStoreDialectFactory.acceptsURL()` so bracketed IPv6 hosts without an explicit port are no longer misparsed as an invalid port
- kept numeric port validation when the port position is unambiguous
- added regression coverage in both `SingleStoreDialectTest` and `JdbcDialectLoaderTest`, including loader coverage for `jdbc:singlestore://[2001:db8::1]/test`
- adjusted the newer test assertions to the API/JUnit level that this branch actually compiles against

Local verification:
- `./mvnw -pl seatunnel-connectors-v2/connector-jdbc -Dtest=SingleStoreDialectTest,JdbcDialectLoaderTest -DfailIfNoTests=false -Dspotless.check.skip=true test`

I also checked WhaleTunnel 2.6/3.0 locally while doing this pass and did not find a corresponding SingleStore dialect implementation there, so there is no same-issue follow-up to file on that side.

**Review by DanielLeens — COMMENTED** [review]
Thanks for the update. I rechecked the current head after the follow-up commit that fixes the `acceptsURL()` blocker for bracketed IPv6 hosts. The dialect fix and the added regression coverage in `SingleStoreDialectTest` and `JdbcDialectLoaderTest` address the previous source-level concern.

The current `Build` check is failing, and this branch is 6 commits behind `dev` (compare status: `diverged`). The failure does not appear to be directly caused by the SingleStore connector changes, so the lowest-cost next step is to sync with the latest `dev` and rerun CI. If it still fails after the sync, please paste the new failing check link and I can help narrow it down.

**Review by DanielLeens — COMMENTED** [review]
Thanks for the update, and thank you for continuing to polish this as a first contribution. I re-reviewed the latest head `76cbf95b74c0` from the full SingleStore dialect path.

The previous Daniel blocker is fixed for me now. The latest `SingleStoreDialectFactory.acceptsURL()` handles bracketed IPv6 hosts correctly, and the regression is covered both at the dialect factory level and through `JdbcDialectLoaderTest`.

# What this PR fixes
- User pain: SingleStore users currently have to rely on generic/MySQL-compatible JDBC behavior without a clear SeaTunnel dialect identity.
- Fix approach: this PR adds a SingleStore JDBC dialect/factory, registers the driver dependency, and adds English/Chinese source and sink documentation.
- One-line summary: the SingleStore dialect path now looks source-ready to me; the remaining merge gate is CI.

# Runtime chain I rechecked
```text
JDBC source/sink initialization
  -> JdbcDialectLoader.load(url, dialect, fieldIde)
      -> ServiceLoader<JdbcDialectFactory>
          -> SingleStoreDialectFactory.acceptsURL(url)
              -> require jdbc:singlestore: prefix
              -> extract host segment
              -> validate empty host / bracketed IPv6 / unambiguous port
          -> SingleStoreDialectFactory.create(...)
              -> new SingleStoreDialect(...)

SingleStore dialect behavior
  -> SingleStoreDialect extends MysqlDialect
      -> dialectName() returns SingleStore
      -> getRowConverter() returns a MySQL-compatible converter that reports SingleStore
      -> upsert, split, type conversion, and default JDBC parameters are inherited from MysqlDialect

Regression coverage
  -> SingleStoreDialectTest covers bracketed IPv6 with and without port
  -> JdbcDialectLoaderTest covers jdbc:singlestore://[2001:db8::1]/test
```

# Findings
- I do not see a new source-level blocker on the current head.
- The earlier bracketed IPv6 URL issue is addressed.
- The new tests are deterministic unit/loader tests; I did not see fixed sleeps, containers, random ports, or external-service dependencies in the added SingleStore tests.
- English and Chinese source/sink docs are both present and aligned with the code path I checked.

# CI
- The current Apache `Build` check is still red: https://github.com/apache/seatunnel/runs/86686986833
- The fork run shows failures in lanes such as `all-connectors-it-1`, `all-connectors-it-7`, and `engine-k8s-it`, with some later lanes cancelled.
- The branch is not behind current `dev` (`ahead_by=12`, `behind_by=0`, compare status `ahead`), so this is no longer the earlier “sync latest dev first” situation.

### Conclusion: can merge after fixes

1. Blocking items
- No source-level blocker from Daniel's side on the current head.
- The latest `Build` must be green before merge.

2. Suggested non-blocking follow-up
- Consider normalizing the PR title module to the common `[Feature][Connector-V2] ...` style, but this is not a source blocker.

Overall, nice progress. From the code-review side, the SingleStore dialect implementation and regression coverage look ready; the remaining step is to get the latest CI green.

**Review by DanielLeens — COMMENTED** [review]
Thanks for the update. I re-reviewed the complete current diff and the SingleStore change and latest CI-only update.

I do not see a source-side blocker on the current head. The current unit-test and engine-v2 lanes remain red, and no SingleStore-specific failure has been isolated, so CI remains outstanding.

Daniel's repository permission is read-only, so this `COMMENTED` review records the technical conclusion only; it does not change GitHub's approval summary. A write-capable maintainer may still need to complete the repository approval gate.

**Review by DanielLeens — COMMENTED** [review]
The SingleStore dialect path remains source-clear: URL matching handles bracketed IPv6, the dialect is registered through the JDBC loader, and the English/Chinese source and sink docs match the current implementation.

GitHub is currently reporting merge conflicts with the latest `dev`, and the required Build is failing. Please resolve the conflicts and rerun the checks. The current technical verdict is `blocking` on the conflict gate, not on a new SingleStore source issue.

**Comment by DanielLeens** [comment]
Thanks for the update. GitHub is currently reporting merge conflicts between this branch and the latest `dev`. Could you please sync with the latest `dev` and resolve the conflicts first? Once that is done and the checks are rerun, I am happy to take another look.

**Comment by DanielLeens** [comment]
Thanks for your contribution for Apache SeaTunnel. 

To facilitate focused communication among contributors, we’ve created a brand-new dedicated #contributors channel in official Slack workspace.
Slack workspace link: https://s.apache.org/seatunnel-slack

This channel is designed for technical discussions, feature design brainstorming, roadmap sync and direct communication with project core Contributors. Everyone who contributes code, docs, test cases or community ideas is warmly invited to join, share thoughts and collaborate together under The Apache Way. Welcome to join us😄

**Comment by davidzollo** [comment]
Thanks for your contribution! This is a solid addition to the project, and your attention to detail really shows. Contributors like you are what keep this community moving forward. Feel free to keep the PRs coming — we're always glad to have you here!

Please ignore the following message if you have joined in the Slack~
```
To facilitate focused communication among contributors, we’ve created a brand-new dedicated #contributors channel in official Slack workspace.
Slack workspace link: https://s.apache.org/seatunnel-slack

This channel is designed for technical discussions, feature design brainstorming, roadmap sync and direct communication with project core Contributors. Everyone who contributes code, docs, test cases or community ideas is warmly invited to join, share thoughts and collaborate together under The Apache Way. Welcome to join 😄
```
